### PR TITLE
Add keyType to postgres s3 helm template

### PR DESCRIPTION
**Pull request by jyanez900**
keyType was missing from the list of s3 config fields.

### CODE REVIEW

**Comment by toscott** [comment]
This change allows us to use role based access when running within EKS.

**Comment by benjaminjb** [comment]
You would need to add this to the `values.yaml` too, right?

File: helm/postgres/values.yaml [hunk 1]
```diff
@@ -182,8 +182,10 @@ postgresVersion: 14
 #   region: ""
 #   # key is the S3 key. This is stored in a Secret.
 #   key: ""
-#   # keySecret is the S3 key secret. This is tored in a Secret.
+#   # keySecret is the S3 key secret. This is stored in a Secret.
 #   keySecret: ""
+#   # keyType can be configured to enable IAM integration via AssumeRole
```
Comment (benjaminjb) [inline helm/postgres/values.yaml:187]:
```suggestion
#   # keyType can be configured to enable IAM integration via AssumeRole
#   # For more on that, see the documentation at https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/backups/#using-an-aws-integrated-identity-provider-and-role
```

**Review by benjaminjb — APPROVED** [review]
LGTM, let me just get a backpatch ready for our devel branch

**Pull request merged**